bot/exts/internal_eval/_helpers.py

```diff
@@ -111,17 +110,17 @@ def dependencies(self) -> typing.Dict[str, typing.Any]:
         }
 
     @property
-    def locals(self) -> typing.Dict[str, typing.Any]:
+    def locals(self) -> dict[str, Any]:
         """Return a mapping of names->values needed for evaluation."""
         return {**collections.ChainMap(self.dependencies, self.context_vars, self._locals)}
 
     @locals.setter
-    def locals(self, locals_: typing.Dict[str, typing.Any]) -> None:
+    def locals(self, locals_: dict[str, Any]) -> None:
```

Comment:
Should be rather `NoReturn` or `None`? 🤔 
```suggestion
    def locals(self, locals_: dict[str, Any]) -> NoReturn:
```

Reply:
This method returns `None` as annotated. `NoReturn` is for [functions that never return](https://www.python.org/dev/peps/pep-0484/#the-noreturn-type).

Reply:
@kwzrd where it returns `None`? 👀

Reply:
Functions that have code execution "drop off" the end of the function return None by default.